### PR TITLE
Refactor test-files to typescript

### PR DESCRIPTION
**Pull request by pmogollons**
* Added types
* Removed underscore
* Get the fileName just on time, not one time for each regex

### CODE REVIEW

File: tools/isobuild/test-files.ts [hunk 1]
```diff
@@ -20,11 +19,13 @@ export const APP_TEST_FILENAME_REGEXPS = [
 
 // Given a path to a file in an app (relative to the app root
 // directory), is this file a test file?
-export function isTestFilePath(path) {
-  const splitPath = path.split(pathSep);
+export function isTestFilePath(path: string) {
```
Comment (jamesmillerburgess) [inline tools/isobuild/test-files.ts]:
```suggestion
export function isTestFilePath(path: string): boolean {
```

File: tools/isobuild/test-files.ts [hunk 2]
```diff
@@ -20,11 +19,13 @@ export const APP_TEST_FILENAME_REGEXPS = [
...
+  // We can remove the || '', but pop function declares a string | undefined
+  // return that is not compatible with regex.test function
+  const fileName = path.split(pathSep).pop() || '';
```
Comment (jamesmillerburgess) [inline tools/isobuild/test-files.ts]: Not sure if this is better:
```suggestion
  const splitPath = path.split(pathSep);
  const fileName = splitPath[splitPath.length - 1];
```

**Comment by afrokick** [comment]
@pmogollons could you please sign the CLA?

**Comment by MarceloPrado** [comment]
Hi @pmogollons, are you still open to implement the suggested changes and fix the conflicts? I could give you a hand if needed.

Thanks

**Comment by filipenevola** [comment]
Hi, as the author is not replying here and we have conflicts I'm going to close this for now. Of course, we can re-open if someone wants to proceed.

**Pull request closed without merging**